### PR TITLE
Do not copy the segment since it is already copied by the ArrayBufferWriter

### PR DESCRIPTION
**Pull request by danielmarbach**
Follow up of https://github.com/Azure/azure-sdk-for-net/pull/19951

### CODE REVIEW

**Comment by danielmarbach** [comment]
// cc @JoshLove-msft since you also approved the other PR

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs [hunk 1]
```diff
@@ -814,24 +814,14 @@ private static BinaryData ReadAmqpDataBody(IEnumerable<Data> body)
         ///
         /// <returns>The set of bytes extracted from the <paramref name="data" />.</returns>
         ///
-        private static byte[] GetDataBytes(Data data)
+        private static ReadOnlyMemory<byte> GetDataBytes(Data data)
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs:817]:
```suggestion
private static ReadOnlyMemory<byte> GetDataBytes(Data data) =>
    data.Value switch
    {
        byte[] byteArray => byteArray,
        ArraySegment<byte> segment => segment,
        _ => ReadOnlyMemory<byte>.Empty
    };
```

The nittest of nits...  :smile:

**Review by jsquire — APPROVED** [review]
Thanks, @danielmarbach.   LGTM.   I'm going to hold on merging until Josh weights in on the comments we added to #19951.   I want to be sure that it wasn't left as a copy in Service Bus for a specific reason.

**Comment by danielmarbach** [comment]
I haven't spent a lot of time in the EventHub code base but I think the fundamental difference is that SB works on Annotatedmessages holding the `IEnumerable<ReadOnlyMemory<byte>>` and then later turn that into a `BinaryData` lazily while EventHub eagerly converts into BinaryData but I might have missed something

**Comment by jsquire** [comment]
> I haven't spent a lot of time in the EventHub code base but I think the fundamental difference is that SB works on Annotatedmessages holding the `IEnumerable<ReadOnlyMemory<byte>>` and then later turn that into a `BinaryData` lazily while EventHub eagerly converts into BinaryData but I might have missed something

Seems likely; it may be due to protecting callers from issues if the underlying AMQP model is updated while the BinaryData instance is held.   If so, then this change makes sense in the short term,  until we adopt the raw AMQP models as well.

**Comment by jsquire** [comment]
I'm going to change tact here and merge this in now;  I want to include it in the stress runs for this month's release.   If we're wrong in our speculation, I'll back it out.

**Pull request merged**